src/Nethermind/Nethermind.Evm.Test/Eip6780Tests.cs

```diff
@@ -142,6 +133,56 @@ public void self_destruct_in_same_transaction(long blockNumber, ulong timestamp)
 
             _processor.Execute(createTx, block.Header, NullTxTracer.Instance);
 
+            AssertCodeHash(contractAddress, Keccak.OfAnEmptyString);
+            AssertStorage(new StorageCell(contractAddress, 1), 0);
+
+            TestState.GetBalance(contractAddress).Should().Be(0);
+            TestState.GetBalance(TestItem.PrivateKeyB.Address).Should().Be(99.Ether());
+        }
+
+        [TestCase(MainnetSpecProvider.GrayGlacierBlockNumber, MainnetSpecProvider.CancunBlockTimestamp)]
+        public void self_destruct_in_initcode(long blockNumber, ulong timestamp)
+        {
+            TestState.CreateAccount(TestItem.PrivateKeyA.Address, 1000.Ether());
+            TestState.Commit(SpecProvider.GenesisSpec);
+            TestState.CommitTree(0);
+            byte[] contractCode = Prepare.EvmCode
+                .SLOAD(1)
+                .EQ(1)
+                .PushData(17)
+                .Op(Instruction.JUMPI)
+                .SSTORE(1, new byte[] { 0x01 })
+                .JUMP(40)
+                .JUMPDEST()
+                .SELFDESTRUCT(TestItem.PrivateKeyB.Address)
+                .JUMPDEST()
+                .Done;
+            byte[] initCode = Prepare.EvmCode
+                .StoreDataInMemory(0, contractCode)
+                .PushData(contractCode.Length)
+                .PushData(0)
+                .SELFDESTRUCT(TestItem.PrivateKeyB.Address)
+                .Op(Instruction.RETURN)
+                .Done;
+            byte[] salt = new UInt256(123).ToBigEndian();
+            Address createTxAddress = ContractAddress.From(TestItem.PrivateKeyA.Address, 0);
+            Address contractAddress = ContractAddress.From(createTxAddress, salt, initCode);
+            byte[] tx1 = Prepare.EvmCode
+                .Create2(initCode, salt, 99.Ether())
```

Comment:
so InitCode, but we're still using Create2? Shouldn't you pass this code as Data?

Reply:
actually, I can't find test cases with trying to selfdestruct as tx.Data argument

Reply:
Fixed